python/ray/dag/context.py

```diff
@@ -58,6 +60,10 @@ class DAGContext:
             executions is beyond the DAG capacity, the new execution would
             be blocked in the first place; therefore, this limit is only
             enforced when it is smaller than the DAG capacity.
+        max_inflight_executions: The maximum number of in-flight executions
+            that can be submitted before consuming the output.
+        overlapping_factor: Determines the degree to which the DAG execution
+            can overlap communication and computation.
```

Comment:
Unclear what unit degree is measured in. Prefer to keep this simple and just make it a bool for now.

Reply:
+1. I think we will eventually decide how many recv and send to overlap to control memory usage. But we can expose that later